vapid/soa.h

```diff
@@ -108,19 +107,29 @@ namespace vapid {
 
         template <size_t col_idx, typename C>
         void sort_by_field(C&& comparator) {
-            reset_sort_reference();
+            size_t num_elems =  size();
+            std::vector<size_t> sort_order_tmp(num_elems);
+            std::vector<size_t> sort_order_reference(num_elems);
```

Comment:
I dislike having to allocate here. My use case will do multiple sorts in between inserts so it would be nice to keep the buffer around.

Reply:
I just made these vectors `thread_local`  so basically they are static now.